src/modules/assisted_release/assisted_release.cpp

```diff
@@ -0,0 +1,312 @@
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "assisted_release.hpp"
+
+#include <px4_platform_common/getopt.h>
+#include <px4_platform_common/log.h>
+#include <px4_platform_common/posix.h>
+
+#include <uORB/Publication.hpp>
+#include <drivers/drv_hrt.h>
+
+#include <uORB/topics/parameter_update.h>
+#include <uORB/topics/sensor_combined.h>
+
+#include <uORB/topics/vehicle_status.h>
+#include <uORB/topics/actuator_controls.h>
+#include <uORB/topics/airspeed.h>
+#include <uORB/topics/rpm.h>
+
+
+int AssistedRelease::print_status()
+{
+	PX4_INFO("Running");
+    PX4_INFO("State: frequency=%d, airspeed=%d, throttle=%d, switch=%d", _rpm.indicated_frequency_rpm > (float)_param_min_rpm.get(), _airspeed.indicated_airspeed_m_s > (float)_param_min_aspd.get(), _throttle >= 0.5f, _rc_channel >= 0.5f);
+
+	return 0;
+}
+
+int AssistedRelease::custom_command(int argc, char *argv[])
+{
+	/*
+	if (!is_running()) {
+		print_usage("not running");
+		return 1;
+	}
+
+	// additional custom commands can be handled like this:
+	if (!strcmp(argv[0], "do-something")) {
+		get_instance()->do_something();
+		return 0;
+	}
+	 */
+
+	return print_usage("unknown command");
+}
+
+
+int AssistedRelease::task_spawn(int argc, char *argv[])
+{
+	_task_id = px4_task_spawn_cmd("assisted_release",
+				      SCHED_DEFAULT,
+				      SCHED_PRIORITY_DEFAULT,
+				      1024,
+				      (px4_main_t)&run_trampoline,
+				      (char *const *)argv);
+
+	if (_task_id < 0) {
+		_task_id = -1;
+		return -errno;
+	}
+
+	return 0;
+}
+
+AssistedRelease *AssistedRelease::instantiate(int argc, char *argv[])
+{
+	int example_param = 0;
+	bool example_flag = false;
+
+	AssistedRelease *instance = new AssistedRelease(example_param, example_flag);
+
+	if (instance == nullptr) {
+		PX4_ERR("alloc failed");
+	}
+
+	return instance;
+}
+
+AssistedRelease::AssistedRelease(int example_param, bool example_flag)
+	: ModuleParams(nullptr)
+{
+}
+
+void AssistedRelease::run()
+{
+
+	// int parameter_update_sub = orb_subscribe(ORB_ID(parameter_update));
+
+	_rpm_sub = orb_subscribe(ORB_ID(rpm));
```

Comment:
Suggestion (optional), consider exclusively using the C++ uORB api, it's actually faster, cheaper, and harder to misuse.